### PR TITLE
Add missing order messages, close #962

### LINKED ISSUE #962: Add missing order messages and update actions

### PR DESCRIPTION
**Pull request by ccsalazarr-commercetools**

### CODE REVIEW

File: commercetools-models/src/main/java/io/sphere/sdk/orders/messages/CustomLineItemStateTransitionMessage.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+import java.time.ZonedDateTime;
+
+/**
+ * This message is the result of the {@link io.sphere.sdk.orders.commands.updateactions.TransitionCustomLineItemState} update action.
+ *
+ * {@include.example io.sphere.sdk.orders.commands.OrderUpdateCommandIntegrationTest#transitionCustomLineItemState()}
+ *
+ */
+@JsonDeserialize(as = CustomLineItemStateTransitionMessage.class)//important to override annotation in Message class
+public class CustomLineItemStateTransitionMessage extends LikeLineItemStateTranstition {
```
Comment (yanns) [inline commercetools-models/src/main/java/io/sphere/sdk/orders/messages/CustomLineItemStateTransitionMessage.java]: `final`?

File: commercetools-models/src/main/java/io/sphere/sdk/orders/messages/LineItemStateTransitionMessage.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+import java.time.ZonedDateTime;
+
+/**
+ * This message is the result of the {@link io.sphere.sdk.orders.commands.updateactions.TransitionLineItemState} update action.
+ *
+ * {@include.example io.sphere.sdk.orders.commands.OrderUpdateCommandIntegrationTest#transitionLineItemState()}
+ *
+ */
+@JsonDeserialize(as = LineItemStateTransitionMessage.class)//important to override annotation in Message class
+public class LineItemStateTransitionMessage extends LikeLineItemStateTranstition {
```
Comment (yanns) [inline commercetools-models/src/main/java/io/sphere/sdk/orders/messages/LineItemStateTransitionMessage.java]: `final`?

File: commercetools-models/src/main/java/io/sphere/sdk/orders/messages/LikeLineItemStateTranstition.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+import java.time.ZonedDateTime;
+
+public class LikeLineItemStateTranstition extends GenericMessageImpl<Order> {
+
+    protected final ZonedDateTime transitionDate;
+    protected final Long quantity;
+    protected final Reference<State> fromState;
+    protected final Reference<State> toState;
+
+    public LikeLineItemStateTranstition(final String id, final Long version, final ZonedDateTime createdAt, final ZonedDateTime lastModifiedAt, final JsonNode resource, final Long sequenceNumber, final Long resourceVersion, final String type, final Class<Order> clazz, final ZonedDateTime transitionDate, final Long quantity, final Reference<State> fromState, final Reference<State> toState) {
```
Comment (yanns) [inline commercetools-models/src/main/java/io/sphere/sdk/orders/messages/LikeLineItemStateTranstition.java]: `LineItemStateLikeTransition`?

Comment (schleichardt) [inline commercetools-models/src/main/java/io/sphere/sdk/orders/messages/LikeLineItemStateTranstition.java]: plus this class should be in package scope ànd abstract

so `public class Whatever`becomes `abstract class Whatever`

File: commercetools-models/src/test/java/io/sphere/sdk/orders/commands/OrderUpdateCommandIntegrationTest.java [hunk 4]
```diff
@@ -254,6 +270,28 @@ public void transitionLineItemState() throws Exception {
...
+                    final LineItemStateTransitionMessage lineItemStateTransitionMessage = lineItemStateTransitionMessageOptional.get();
+
+                    final String lineItemIdFromMessage = lineItemStateTransitionMessage.getLineItemId();
+                    assertThat(lineItemIdFromMessage).isEqualTo(order.getLineItems().get(0).getId());
+                    final ZonedDateTime transitionDateFromMessage = lineItemStateTransitionMessage.getTransitionDate();
+                    assertThat(transitionDateFromMessage).isEqualTo(actualTransitionDate);
+                    final Long quantityFromMessage = lineItemStateTransitionMessage.getQuantity();
+                    assertThat(quantityFromMessage).isEqualTo(quantity);
+                    final Reference<State> fromStateFromMessage = lineItemStateTransitionMessage.getFromState();
+                    assertThat(fromStateFromMessage).isEqualTo(Reference.of("state", initialState.getId()));
```
Comment (schleichardt) [inline commercetools-models/src/test/java/io/sphere/sdk/orders/commands/OrderUpdateCommandIntegrationTest.java]: `Reference.of("state", initialState.getId())` could be replaced with `initialState.toReference()`.

Showing the example with the String constant is not the best practice for users.

File: commercetools-models/src/main/java/io/sphere/sdk/orders/messages/LikeLineItemStateTranstition.java (hunk 3 above)
Comment (yanns) [inline commercetools-models/src/main/java/io/sphere/sdk/orders/messages/LikeLineItemStateTranstition.java]: I also though of having this class `abstract` but `GenericMessageImpl` is not `abstract`

**Comment by schleichardt** [comment]
:+1:

File: commercetools-models/src/main/java/io/sphere/sdk/orders/messages/LineItemLikeStateTransitionMessage.java [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+import java.time.ZonedDateTime;
+
+/**
+ * This message is the result of the {@link io.sphere.sdk.orders.commands.updateactions.TransitionLineItemState} update action.
+ *
+ * {@include.example io.sphere.sdk.orders.commands.OrderUpdateCommandIntegrationTest#transitionLineItemState()}
+ *
+ */
+@JsonDeserialize(as = LineItemLikeStateTransitionMessage.class)//important to override annotation in Message class
+public final class LineItemLikeStateTransitionMessage extends LineItemLikeStateTransition {
```
Comment (schleichardt) [inline commercetools-models/src/main/java/io/sphere/sdk/orders/messages/LineItemLikeStateTransitionMessage.java]: @ccsalazarr-commercetools I think you want to name it `LineItemStateTransitionMessage`, it contains a `Like` too much.

**Pull request merged**